test/e2e/helpers_test.go

```diff
@@ -614,3 +681,28 @@ func getPodByLabels(ctx context.Context,
 	}
 	return podList.Items, nil
 }
+
+// scaleDeployment scales the deployment to the given number of replicas and waits for the replicas to be ready.
+func scaleDeployment(ctx context.Context, t *testing.T, env environments.Environment, deployment types.NamespacedName, replicas int32) {
+	t.Helper()
+
+	scale := &autoscalingv1.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: deployment.Namespace,
+			Name:      deployment.Name,
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: replicas,
+		},
+	}
+	_, err := env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx, deployment.Name, scale, metav1.UpdateOptions{})
```

Comment:
nit: `env.Cluster().Client().AppsV1().Deployments(deployment.Namespace)` could be extracted to a variable.